### ISSUE TITLE
Merge Size and Count Inconsistent

### ISSUE THREAD

**Issue by texlnghorn** [elastic/elasticsearch#17462]
**Elasticsearch version**:
1.7.5

**JVM version**:
1.8.0_40-64

**OS version**:
centos6

**Description of the problem including expected versus actual behavior**:
The current merge count is different between _cat/nodes and _stats/merges.  We show multiple merges being performed on the _cat/nodes api, however, _stats/merges shows no merges being performed for any index.  We are not sure what is the true merge activity on the cluster.  Also, is there a way to see what shards are bing merged?

![cat-nodes](https://cloud.githubusercontent.com/assets/6633418/14191693/f8057626-f75e-11e5-9a62-10c0c76de164.png)
![stats-merges](https://cloud.githubusercontent.com/assets/6633418/14191712/0a9e1cf2-f75f-11e5-85f9-27187f099f09.png)

**Steps to reproduce**:
 We are not sure how the cluster gets into this state.  We have experienced this multiple times.  We have not found a way to clear what we think are stuck merges without bouncing the nodes.

**Provide logs (if relevant)**:

**Comment by clintongormley** [comment]
Hi @texlnghorn 

Are you saying that the merges reported by cat-nodes never change? While those from nodes-stats do? The problem is that 1.7.5 is really quite old now.  We're working towards releasing 5.0, and so much has changed since 1.7.5  We'd be interested if you were able to replicate this on a recent version.

**Comment by texlnghorn** [comment]
1.7.5 was released 2/2/2016.  But I know that you are saying that the 1.7.x version is old.  Yes, it appears that merges are stuck querying cat-nodes but not stuck querying nodes-stats.  We have seen the same number of docs and same size reported in cat-nodes for more than 2 weeks.  If we bounce the node, then the merges info goes back to 0.  Eventually we see the same situation again after a few days.  Unfortunately we cannot upgrade to 2.x or 5.x for some time.  I had not seen anyone else report this and was wondering if others had experienced the same thing.   We don't know if we really have stuck merges or if something is not getting reset properly in cat-nodes.

**Comment by clintongormley** [comment]
It sounds very much like an issue with stats reporting, but curious that it happens in cat-nodes and not in nodes-stats. I haven't heard other reports of this but that doesn't mean that it doesn't happen or that it has already been fixed.  Worth investigating I think

**Comment by mkelkarbv** [comment]
We see the same issue on version 1.7.3. the _stats/merges reports no current merges, where as _cat/nodes API reports 3 on going merges on different nodes.

**Comment by s1monw** [comment]
@mkelkarbv can you provide a sample output what doesn't match? I can't see any example like a json snippet and a CAT line that is wrong.. hard to tell what's going on?!

**Comment by thebearmayor** [comment]
@s1monw I can provide that.

```
# curl -s 'localhost:9200/_cat/nodes?v&h=id,mc,mcd'
id   mc     mcd
VXvB  0       0
lnIR  0       0
_oF8  0       0
bfHD  0       0
sv91  2 1265846
EtuE  0       0
uPsj  0       0
TSPN  0       0
faYE  0       0
_K5Y  0       0
nz9W  0       0
piz6  1  347959
R6jG  0       0
QP9S  0       0
kG31  0       0
rncC  0       0
0WSl  0       0
-mEf  0       0
1zgM  0       0
```
```
curl -s 'localhost:9200/_cat/indices?v&h=i,mc,mcd'
i                      mc mcd
&part-reviewpartition7  0   0
&part-reviewpartition2  0   0
&part-reviewpartition6  0   0
&part-reviewpartition5  0   0
&part-reviewpartition8  0   0
&part-reviewpartition4  0   0
&part-reviewpartition9  0   0
percolators             0   0
&part-reviewpartition0  0   0
&part-reviewpartition3  0   0
&part-reviewpartition1  0   0
```
These were taken at the same time. The 3 merges that show when we query _cat/nodes never go away. The info from _cat/indices seems to be correct -- merges come and go but currently there are none.

**Comment by s1monw** [comment]
this is fixed by #13801 in 2.0 upwards.... The node stats include the stats of all shards ever on that node even the old ones. Yet, it was also counting the current stats not just totals. this is now fixed and this is essentially a duplicate of  #13386  

the stats in `curl -s 'localhost:9200/_cat/indices?v&h=i,mc,mcd'` are accurate in 1.x
